### PR TITLE
Topic algorithms- exposure(rescale_intensity)

### PR DESCRIPTION
**Pull request by rnvikram**
Implemented the rescale intensity algorithm from the transform module with TensorFlow. The driver file can be run to test the algorithm. rescale_intensity from rescale_intensity must be called to use the algorithm.

### CODE REVIEW

File: transform/rescale_intensity/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Rescale Intensity
```
Comment (shakes76) [inline transform/rescale_intensity/README.md]: Good, but not dependencies list

File: transform/rescale_intensity/driver.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import os
```
Comment (shakes76) [inline transform/rescale_intensity/driver.py]: Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver/module does.

File: transform/rescale_intensity/driver.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        tf_output = rescale_intensity(array, in_range=in_range)
+        np_output = rescale_intensity_numpy(array, in_range=in_range)
+        result = np.array_equal(np_output, tf_output)
+
+    if in_range == None and out_range != None:
+        tf_output = rescale_intensity(array, out_range=out_range)
+        np_output = rescale_intensity_numpy(array, out_range=out_range)
+        result = np.array_equal(np_output, tf_output)
+
+    if in_range != None and out_range != None:
```
Comment (shakes76) [inline transform/rescale_intensity/driver.py]: Good to add inline comments even if original code didn't have them

File: transform/rescale_intensity/rescale_intensity.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      pass
+  elif out_range in DTYPE_RANGE:
+      pass
+  else:
+      raise ValueError('Unsupported input to out_range', out_range)
+  input_image = tf.constant(input_image)
+  sess = tf.InteractiveSession()
+  tf.global_variables_initializer().run()
+  imin, imax = intensity_range(input_image,dtype, in_range)
+  omin, omax = intensity_range(input_image,dtype,out_range,clip_negative=(imin >= 0))
```
Comment (shakes76) [inline transform/rescale_intensity/rescale_intensity.py]: More inline comments please

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
OK
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
## Commit Log
Meaningful commit messages 
Progressive commits OK, more details next time
## Documentation
Description and Comments  good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required, changes to hard coded directories**
Request Description minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info. Docstrings are important too.

File: transform/rescale_intensity/rescale_intensity.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    intensity_range([-5,5,10],float,range_values='image',clip_negative=True)  -> 0,10
+
+    """
+
+
+    if range_values == 'dtype':
+        range_values = dtype
+    if str(range_values) == 'image':
+        i_min = tf.reduce_min(image).eval()
+        i_max = tf.reduce_max(image).eval()
```
Comment (shakes76) [inline transform/rescale_intensity/rescale_intensity.py]: Good to add inline comments even if original code didn't have them

**Comment by rnvikram** [comment]
Hello Shakes,
Thanks for reviewing the request.

1. **Commit log**- I had initially run all commits in the master branch. You can access it from here. https://github.com/rnvikram/PatternFlow/tree/master/rescale%20tensor

2. > Remove references to hard coded directories. Why is Kirti's directory in the code?
I used collab to implement the module. Since my machine did not have TensorFlow I used Kirti's machine to test the driver script. I had to make some changes. Looks like missed out removing the directory when transferring the code to my machine

3. > Good, but not dependencies list
I'm not sure about the comment. Does the feedback mean that I would have to include the dependencies used in the readme file?

4. > Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver/module does.
I've added the header to the tester file. Not sure if it should be added to the implementation. 

5. > Good to add inline comments even if original code didn't have them
Added inline comments as requested

6. > Good to add inline comments even if original code didn't have them
Added inline comments as requested


Need help with:
> Good, but not dependencies list
> Remove references to hard coded directories

Thanks

**Pull request merged**